### PR TITLE
(Revised) - [HacktoberFest2023]: Added cpp.chat and No Diagnostic Required podcast under C++ podcasts section

### PR DESCRIPTION
**Pull request by honey-speaks-tech**
## What does this PR do?
Add resources

### Description
- Added link to cpp.chat podcast
- Added link to No Diagnostic Required podcast

### Why is this valuable (or not)?
- cpp.chat and No Diagnostic Required are great C++ podcasts which brings some C++ news, interviews with C++ community members etc

### How do we know it's really free?
- Visit websites [cpp.chat](https://cpp.chat/) & [No Diagnostic Required](https://nodiagnosticrequired.tv/)

### For book lists, is it a book? For course lists, is it a course? etc.
- N.A

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction).
- [X] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by Ashu104** [comment]
hey @honey-speaks-tech 
you haven't passed some check on your commit because of avoiding the alphabetical ordering and remove trailing slash on your commit. try to change them. 
you can refer. -->
   63:3-63:61  warning  Remove trailing slash (https://nodiagnosticrequired.tv/)  no-url-trailing-slash  remark-lint
   62:3-62:32  warning  Remove trailing slash (https://cpp.chat/)                 no-url-trailing-slash  remark-lint
   1-63:118  warning  Alphabetical ordering: swap l.62 and l.61                alphabetize-lists      remark-lint

File: casts/free-podcasts-screencasts-en.md [hunk 1]
```diff
@@ -59,6 +59,8 @@
 * [C++ STL by example](https://www.youtube.com/playlist?list=PLZ9NgFYEMxp5oH3mrr4IlFBn03rjS-gN1) - Douglas Schmidt (screencast)
 * [C++ STL: The ONLY Video You Need | Compulsory for DSA/CP](https://www.youtube.com/watch?v=PZogbfU4X5E) - Utkarsh Gupta (screencast)
 * [CppCast](http://cppcast.com) - Conor Hoekstra, Jason Turner, JeanHeyd Meneide, Matt Godbolt, Rob Irving (podcast)
+* [cpp.chat](https://cpp.chat/) - Jon Kalb, Phil Nash (podcast)
+* [No Diagnostic Required](https://nodiagnosticrequired.tv/) - Anastasia Kazakova, Phil Nash (podcast & YouTube show)
```
Comment (eshellman) [inline casts/free-podcasts-screencasts-en.md:63]:
```suggestion
* [No Diagnostic Required](https://nodiagnosticrequired.tv/) - Anastasia Kazakova, Phil Nash (podcast
```

**Review by Ajay-singh1 — APPROVED** [review]
LGTM!

**Comment by honey-speaks-tech** [comment]
@eshellman @Ashu104 Fixed the lint issue in this PR. But an issue blocking the merge which was already reported by someone here in the issue [10557](https://github.com/EbookFoundation/free-programming-books/issues/10557)
I have skimmed through this file, but unable to understand what is to be done as it is in a different language. So waiting until the issue is fixed. I have commented on the issue as a "show-stopper" for many PRs.

**Comment by eshellman** [comment]
linter issue fixed, closed/opened to rebuild

**Pull request merged**